drivers/d3d12/d3d12_context.cpp

```diff
@@ -501,7 +493,13 @@ void D3D12Context::_dump_adapter_info(int p_index) {
 }
 
 Error D3D12Context::_create_device(DeviceBasics &r_basics) {
-	HRESULT res = D3D12CreateDevice(gpu.Get(), D3D_FEATURE_LEVEL_11_0, IID_PPV_ARGS(r_basics.device.GetAddressOf()));
+	HRESULT res;
+	if (r_basics.factory) {
+		r_basics.factory->SetFlags(D3D12_DEVICE_FACTORY_FLAG_ALLOW_RETURNING_EXISTING_DEVICE | D3D12_DEVICE_FACTORY_FLAG_ALLOW_RETURNING_INCOMPATIBLE_EXISTING_DEVICE);
```

Comment:
Why these flags? I'm wondering if re-using a device instance can be somehow detrimental for local devices.

Reply:
This might be unnecessary, I'll check if it's working without it (flags are copy-pasted from Mesa).